.cursor/rules/30-typescript.mdc

```diff
@@ -0,0 +1,14 @@
+---
+name: TypeScript scripts
+description: TS scripting, helpers, lint/tests
+globs:
+  - 'script/**/*.ts'
+  - 'tasks/**/*.ts'
+---
+
+- Follow [CONV:SCRIPTS-TS] for TS script conventions; run with `bunx tsx`; obey `.eslintrc.cjs`; avoid `any`; use TypeChain types (e.g., `ILiFi.BridgeDataStruct`).
+- Always reuse existing helper functions from `script/utils/`, `script/demoScripts/utils/`, and other helper directories; prefer existing patterns over new ones. Search for existing helpers before implementing new functionality.
+- Do NOT use deprecated ethers-based helpers (`getProvider`, `getWalletFromPrivateKeyInDotEnv`, ethers `sendTransaction`, `ensureBalanceAndAllowanceToDiamond`); use viem-based flows instead.
+- New helpers require colocated `{name}.test.ts` with 100% coverage using Bun (`describe`/`it`/`expect`); cover edge cases, type safety, and error paths.
+- CLI: use `citty`; logging via `consola`; validate env via `getEnvVar()`; exit 0/1 appropriately.
+- After TS changes run lint/tests (or state which remain) to ensure eslint/type checks pass.
```

Comment:
probably it's work to mention that if we will prompt 'demo script' then we mean script that it's inside 'script/demoScripts' and that usually they follow similar patterns in terms of how they are structured. And we can give references to demoLidoWrapper.ts, demoUnit.ts, demoEco.ts